packages/scandipwa/src/component/CategoryPagination/CategoryPagination.component.js

```diff
@@ -62,50 +73,22 @@ export class CategoryPagination extends PureComponent {
 
     renderPageLinks() {
         const {
-            totalPages,
-            paginationFrame,
-            paginationFrameSkip,
-            currentPage
+            currentPage,
+            firstFramePage,
+            lastFramePage
         } = this.props;
 
-        let pages = [];
+        const pages = [];
         let i;
 
-        // Render next pagination links
-        for (i = currentPage; i <= currentPage + paginationFrame; i++) {
-            if (i <= totalPages && pages.length <= paginationFrameSkip) {
-                pages.push(this.renderPageLink(
-                    i,
-                    __('Page %s', i),
-                    i.toString(),
-                    i === currentPage
-                ));
-            }
-        }
-
-        // Render previous pagination links if necessary
-        for (i = 1; i < currentPage; i++) {
-            if (pages.length < paginationFrame) {
-                const id = currentPage - i;
-                const pageData = this.renderPageLink(
-                    id,
-                    __('Page %s', id),
-                    id.toString()
-                );
-
-                pages = [pageData, ...pages];
-            }
-        }
-
-        // Edge case for rendering correct count of next links when current page is 1
-        if (currentPage === 1 && pages.length < totalPages) {
-            for (i = pages.length + 1; i <= paginationFrame; i++) {
-                pages.push(this.renderPageLink(
-                    i,
-                    __('Page %s', i),
-                    i.toString()
-                ));
-            }
+        // Render frame pagination links
+        for (i = firstFramePage; i <= lastFramePage; i++) {
```

Comment:
Can we change this loop to reduce loop?

Reply:
I got rid of loop (yet it looked like this was exactly the perfect case for loops).
I added `range` method to create an array from interval of numbers (like in Python) and then applied `map()` to it.

But I do really need an explanation why we enforce not using `let` and `loops` at all. I can think of some reasons (like no exit condition for loop or unexpected change of let variable), but there should be more.